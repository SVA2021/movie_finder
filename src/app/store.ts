import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import authReducer from '../features/auth/authSlice';
import movieFinderReducer from '../features/movieFinder/movieFinderSlice';

export const store = configureStore({
  reducer: {
    auth: authReducer,
		movieFinder: movieFinderReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
