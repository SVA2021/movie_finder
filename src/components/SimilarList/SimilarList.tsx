import {Box, Typography} from "@mui/material";
import {FC, memo, useEffect} from 'react';
import {useParams} from "react-router-dom";
import {SwipeListTemplate} from '..';
import {useAppDispatch, useAppSelector} from "../../app/hooks";
import {selectDetails} from '../../features/movieFinder/movieFinderSlice';
import {getMovieSimilarAsync} from "../../features/movieFinder/movieFinderThunks";

export const SimilarList: FC = memo(() => {

  const dispatch = useAppDispatch();
  const DetailsPage = useAppSelector(selectDetails);
  const SimilarFilms = DetailsPage.similars;
  const params = useParams();
  const id = Number(params.id)

  useEffect(() => {
    dispatch(getMovieSimilarAsync(id));
  }, [id]);

  return (
    <Box pt={{xs: 2, lg: 4}} color={'common.white'} >
      <Typography ml={{lg: 15, xl: 20}} gutterBottom variant="h3" component={'h3'}>
        Похожее
      </Typography>
      <SwipeListTemplate data={SimilarFilms?.data ?? []} />
    </Box>
  );
});