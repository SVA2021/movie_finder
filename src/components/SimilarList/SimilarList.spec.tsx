import {render, screen} from "@testing-library/react";
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';
import {store} from "../../app/store";
import {SimilarList} from "./SimilarList";

describe('render SimilarList', () => {

  const setUp = () => render(
    <Provider store={store}>
      <BrowserRouter>
        <SimilarList />
      </BrowserRouter>
    </Provider>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('similar-list')).toMatchSnapshot();
  });

  it('check text', () => {
    setUp();
    expect(screen.getByText(/Похожее/i)).toBeInTheDocument();
  });

})