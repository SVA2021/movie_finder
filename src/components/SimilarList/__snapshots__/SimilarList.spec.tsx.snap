// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SimilarList snapshot test 1`] = `
<div
  class="MuiBox-root css-1baeknb"
  data-testid="similar-list"
>
  <h3
    class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-hjvxqc-MuiTypography-root"
  >
    Похожее
  </h3>
  <div
    class="MuiGrid-root MuiGrid-container css-1nsfak8-MuiGrid-root"
    data-testid="swipe-list-template"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-h3 css-13mkr37-MuiTypography-root"
        data-testid="empty-data-box"
      >
        Нет данных
      </p>
    </div>
  </div>
</div>
`;
